### ISSUE TITLE
Regexp::Grammars broken since 5.17.1

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#12787]
Migrated from [rt.perl.org#116823](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=116823) (status was 'resolved')

Searchable as RT116823$

**Comment by p5pRT** [comment]
### From andrew@cleverdomain.org

#### Created by andrew@cleverdomain.org

This is a bug report for perl from andrew@&#8203;cleverdomain\.org\,
generated with the help of perlbug 1\.39 running under perl 5\.17\.9\.


\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
Regexp&#8203;::Grammars fails all of its tests beginning with perl 5\.17\.1 due to
the reworking of \(?\{\.\.\.\}\) and \(??\{\.\.\.\}\) in that version\. The tests all fail
with "Eval\-group not allowed at runtime\, use re 'eval'" and real code using
Regexp&#8203;::Grammars is expected to fail in the same way\.

Regexp&#8203;::Grammars works by using overload&#8203;::constant qr to bless all qr's in
its scope into objects that have overloaded qq\, which then does all kinds of
string replacements on the pattern to turn it into one brimming with
\(??\{\.\.\.\}\) constructs\. In previous versions of perl\, this worked fine
\(possibly because all of the overload mojo wasn't considered "runtime"\)\, but
now it's broken\.

I'm not sure if this is to be considered a bug in perl\, or if the \*previous\*
behavior was a bug\. If this is not a perl bug\, how can Regexp&#8203;::Grammars fix
itself without requiring its users to pepper their code with use re 'eval'?

Thanks\,

Andrew

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl 5.17.9:

Configured by andrew at Sun Feb 17 02:44:05 EST 2013.

Summary of my perl5 (revision 5 version 17 subversion 9) configuration:
  Derived from: 480c67241f1595db244990ae2327660f1eec3602
  Platform:
    osname=linux, osvers=3.5.0-23-generic, archname=x86_64-linux
    uname='linux goron 3.5.0-23-generic #35-ubuntu smp thu jan 24
13:15:40 utc 2013 x86_64 x86_64 x86_64 gnulinux '
    config_args='-de
-Dprefix=/home/andrew/perl5/perlbrew/perls/perl-5.17.8 -Dusedevel
-Aeval:scriptdir=/home/andrew/perl5/perlbrew/perls/perl-5.17.8/bin'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=define, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-fno-strict-aliasing -pipe -fstack-protector
-I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2',
    cppflags='-fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.7.2', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t',
lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -fstack-protector -L/usr/local/lib'
    libpth=/usr/local/lib /lib/x86_64-linux-gnu /lib/../lib
/usr/lib/x86_64-linux-gnu /usr/lib/../lib /lib /usr/lib
    libs=-lnsl -ldl -lm -lcrypt -lutil -lc
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lc
    libc=, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.15'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
    cccdlflags='-fPIC', lddlflags='-shared -O2 -L/usr/local/lib
-fstack-protector'

Locally applied patches:



@INC for perl 5.17.9:
    /home/andrew/perl5/perlbrew/perls/perl-5.17.8/lib/site_perl/5.17.9/x86_64-linux
    /home/andrew/perl5/perlbrew/perls/perl-5.17.8/lib/site_perl/5.17.9
    /home/andrew/perl5/perlbrew/perls/perl-5.17.8/lib/5.17.9/x86_64-linux
    /home/andrew/perl5/perlbrew/perls/perl-5.17.8/lib/5.17.9
    .


Environment for perl 5.17.9:
    HOME=/home/andrew
    LANG=en_US.UTF-8
    LANGUAGE=
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/andrew/perl5/perlbrew/bin:/home/andrew/perl5/perlbrew/perls/perl-5.17.8/bin:/usr/lib/lightdm/lightdm:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
    PERLBREW_BASHRC_VERSION=0.58
    PERLBREW_HOME=/home/andrew/.perlbrew
    PERLBREW_MANPATH=/home/andrew/perl5/perlbrew/perls/perl-5.17.8/man
    PERLBREW_PATH=/home/andrew/perl5/perlbrew/bin:/home/andrew/perl5/perlbrew/perls/perl-5.17.8/bin
    PERLBREW_PERL=perl-5.17.8
    PERLBREW_ROOT=/home/andrew/perl5/perlbrew
    PERLBREW_VERSION=0.58
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From arodland@cpan.org

The commit message for e03b874abb33f381913aa76467fee51dbdc78263 makes me 
think that Regexp&#8203;::Grammars ought to work without changes\, since a change 
was made for a similar overload\-related breakage in Regexp&#8203;::Common\.

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

The commit message for e03b874abb33f381913aa76467fee51dbdc78263 makes me 
think that Regexp&#8203;::Grammars ought to work without changes\, since a change 
was made for a similar overload\-related breakage in Regexp&#8203;::Common\.

**Comment by p5pRT** [comment]
### From @jkeenan

On Sun Feb 17 22&#8203;:07&#8203;:37 2013\, andrew@&#8203;cleverdomain\.org wrote&#8203;:
> This is a bug report for perl from andrew@&#8203;cleverdomain\.org\,
> generated with the help of perlbug 1\.39 running under perl 5\.17\.9\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> Regexp&#8203;::Grammars fails all of its tests beginning with perl 5\.17\.1 due
> to
> the reworking of \(?\{\.\.\.\}\) and \(??\{\.\.\.\}\) in that version\. The tests all
> fail
> with "Eval\-group not allowed at runtime\, use re 'eval'" and real code
> using
> Regexp&#8203;::Grammars is expected to fail in the same way\.
> 
> Regexp&#8203;::Grammars works by using overload&#8203;::constant qr to bless all
> qr's in
> its scope into objects that have overloaded qq\, which then does all
> kinds of
> string replacements on the pattern to turn it into one brimming with
> \(??\{\.\.\.\}\) constructs\. In previous versions of perl\, this worked fine
> \(possibly because all of the overload mojo wasn't considered
> "runtime"\)\, but
> now it's broken\.
> 
> I'm not sure if this is to be considered a bug in perl\, or if the
> \*previous\*
> behavior was a bug\. If this is not a perl bug\, how can
> Regexp&#8203;::Grammars fix
> itself without requiring its users to pepper their code with use re
> 'eval'?
> 
> Thanks\,
> 
> Andrew

We need to investigate the issues which Andrew has raised\, as this
sounds like a regression and\, hence\, a blocker for 5\.18\.

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @iabyn

On Sun\, Feb 17\, 2013 at 10&#8203;:07&#8203;:37PM \-0800\, Andrew Rodland wrote&#8203;:
> Regexp&#8203;::Grammars fails all of its tests beginning with perl 5\.17\.1 due to
> the reworking of \(?\{\.\.\.\}\) and \(??\{\.\.\.\}\) in that version\. The tests all fail
> with "Eval\-group not allowed at runtime\, use re 'eval'" and real code using
> Regexp&#8203;::Grammars is expected to fail in the same way\.
> 
> Regexp&#8203;::Grammars works by using overload&#8203;::constant qr to bless all qr's in
> its scope into objects that have overloaded qq\, which then does all kinds of
> string replacements on the pattern to turn it into one brimming with
> \(??\{\.\.\.\}\) constructs\. In previous versions of perl\, this worked fine
> \(possibly because all of the overload mojo wasn't considered "runtime"\)\, but
> now it's broken\.
> 
> I'm not sure if this is to be considered a bug in perl\, or if the \*previous\*
> behavior was a bug\. If this is not a perl bug\, how can Regexp&#8203;::Grammars fix
> itself without requiring its users to pepper their code with use re 'eval'?

Given that Regexp/Grammars\.pm is 6000\+ lines of code\, is it possible to
supply a short test script \(with no external dependencies\) that uses
overload&#8203;::constant qr and overloaded qq in such a way that demonstrates
the changed behaviour?

Thanks\.

\-\- 
No man treats a motor car as foolishly as he treats another human being\.
When the car will not go\, he does not attribute its annoying behaviour to
sin\, he does not say\, You are a wicked motorcar\, and I shall not give you
any more petrol until you go\. He attempts to find out what is wrong and
set it right\.
&nbsp;    \-\- Bertrand Russell\,
&nbsp;       Has Religion Made Useful Contributions to Civilization?

**Comment by p5pRT** [comment]
### From arodland@cpan.org

On Sat Mar 02 16&#8203;:38&#8203;:19 2013\, davem wrote&#8203;:
> On Sun\, Feb 17\, 2013 at 10&#8203;:07&#8203;:37PM \-0800\, Andrew Rodland wrote&#8203;:
> > Regexp&#8203;::Grammars fails all of its tests beginning with perl 5\.17\.1
> due to
> > the reworking of \(?\{\.\.\.\}\) and \(??\{\.\.\.\}\) in that version\. The tests
> all fail
> > with "Eval\-group not allowed at runtime\, use re 'eval'" and real
> code using
> > Regexp&#8203;::Grammars is expected to fail in the same way\.
> >
> > Regexp&#8203;::Grammars works by using overload&#8203;::constant qr to bless all
> qr's in
> > its scope into objects that have overloaded qq\, which then does all
> kinds of
> > string replacements on the pattern to turn it into one brimming with
> > \(??\{\.\.\.\}\) constructs\. In previous versions of perl\, this worked fine
> > \(possibly because all of the overload mojo wasn't considered
> "runtime"\)\, but
> > now it's broken\.
> >
> > I'm not sure if this is to be considered a bug in perl\, or if the
> \*previous\*
> > behavior was a bug\. If this is not a perl bug\, how can
> Regexp&#8203;::Grammars fix
> > itself without requiring its users to pepper their code with use re
> 'eval'?
> 
> Given that Regexp/Grammars\.pm is 6000\+ lines of code\, is it possible
> to
> supply a short test script \(with no external dependencies\) that uses
> overload&#8203;::constant qr and overloaded qq in such a way that
> demonstrates
> the changed behaviour?
> 
> Thanks\.
> 

I distilled it down as far as I could in the attached file\. Here's a 
snippet of "perlbrew exec perl rgtest\.pl"&#8203;:

perl\-5\.14\.1
==========
ok 1 \- It matches
ok 2 \- Code ran
1\.\.2


perl\-5\.16\.2
==========
ok 1 \- It matches
ok 2 \- Code ran
1\.\.2


perl\-5\.16\.2@&#8203;ss
==========
ok 1 \- It matches
ok 2 \- Code ran
1\.\.2


perl\-5\.17\.0
==========
ok 1 \- It matches
ok 2 \- Code ran
1\.\.2


perl\-5\.17\.1
==========
Eval\-group not allowed at runtime\, use re 'eval' in regex m/\(?\{ 
$RGP&#8203;::Success = 1 \}\)foo/ at rgtest\.pl line 27\.


perl\-5\.17\.8
==========
Eval\-group not allowed at runtime\, use re 'eval' in regex m/\(?\{ 
$RGP&#8203;::Success = 1 \}\)foo/ at rgtest\.pl line 27\.

**Comment by p5pRT** [comment]
### From arodland@cpan.org

[rgtest.pl](https://rt-archive.perl.org/perl5/Ticket/Attachment/1197651/617575/rgtest.pl)

**Comment by p5pRT** [comment]
### From arodland@cpan.org

On Sat Mar 02 19&#8203;:26&#8203;:20 2013\, arodland wrote&#8203;:
> I distilled it down as far as I could in the attached file\. Here's a 
> snippet of "perlbrew exec perl rgtest\.pl"&#8203;:
> 
> perl\-5\.14\.1
> ==========
> ok 1 \- It matches
> ok 2 \- Code ran
> 1\.\.2
> 
> 
> perl\-5\.16\.2
> ==========
> ok 1 \- It matches
> ok 2 \- Code ran
> 1\.\.2
> 
> 
> perl\-5\.16\.2@&#8203;ss
> ==========
> ok 1 \- It matches
> ok 2 \- Code ran
> 1\.\.2
> 
> 
> perl\-5\.17\.0
> ==========
> ok 1 \- It matches
> ok 2 \- Code ran
> 1\.\.2
> 
> 
> perl\-5\.17\.1
> ==========
> Eval\-group not allowed at runtime\, use re 'eval' in regex m/\(?\{ 
> $RGP&#8203;::Success = 1 \}\)foo/ at rgtest\.pl line 27\.
> 
> 
> perl\-5\.17\.8
> ==========
> Eval\-group not allowed at runtime\, use re 'eval' in regex m/\(?\{ 
> $RGP&#8203;::Success = 1 \}\)foo/ at rgtest\.pl line 27\.

Interestingly\, Adding "use re 'eval'" doesn't suppress the error\, no 
matter which package I add it to \(even if I add it in all three\)\.

**Comment by p5pRT** [comment]
### From @iabyn

On Sat\, Mar 02\, 2013 at 07&#8203;:26&#8203;:21PM \-0800\, Andrew Rodland via RT wrote&#8203;:
> I distilled it down as far as I could in the attached file\. Here's a 
> snippet of "perlbrew exec perl rgtest\.pl"&#8203;:

Thanks for the reduced test case\.

My preliminary analysis is that

1\) the test script \(and thus presumably Regexp&#8203;::Grammars\) only worked
because it was inadvertently exploiting a hole in the "use re eval"
mechanism\, and that perl 5\.17\.x is correctly refusing to execute the code;

2\) but looking at ways in which Regexp&#8203;::Grammars should be modifiable to
"play nicely" aren't in fact working\, which indicates that there are bugs
in the new perl mechanism too\, which I'll need to look into further\, and
presumably fix\.

The basic issue is this line in the  q\{""\} overload method&#8203;:

&nbsp;    return '\(?\{ $RGP&#8203;::Success = 1 \}\)' \. $$re;

this is basically creating an arbitrary string at run\-time\, which is
subsequently interpreted as as a pattern&#8203;: which \*should\* cause the fatal
error\. For example\, modifying that line to&#8203;:

&nbsp;    return '\(?\{ $RGP&#8203;::Success = 1 \}\)' \. $ARGV\[0\] \. $$re;

and when run as&#8203;:

&nbsp;     perl5162 \-w ~/tmp/rgtest\.pl '\(?\{ system "echo hacked"\}\)'

you get&#8203;:

&nbsp;    hacked
&nbsp;    ok 1 \- It matches
&nbsp;    ok 2 \- Code ran
&nbsp;    1\.\.2

which shows that perl is executing code in arbitrary run\-time strings
without 'use re eval' in scope\.

The basic rule for 5\.17\.1 onwards is that literal text containing \(?\{\}\)
that is within a pattern doesn't need 'use re eval'; everything else
should\.  This is in many ways a relaxing of the constraints from earlier
perls&#8203;: this now works without 'use re eval'\, even though it is a run\-time
pattern&#8203;:

&nbsp;    $x = "foo"; $x =~ /\(?\{ print "boo\\n" \}\)$x/;

while other things that were lax have been stiffened\.

However\, I would expect something like the following \*should\* work and be
safe\, but it doesn't&#8203;:

&nbsp;    use re 'eval'; return qr/\(?\{ $RGP&#8203;::Success = 1 \}\)$$re/;

And as you pointed out\,  a 'use re eval' even in the scope of the main
qr/foo/ doesn't turn off the fatal error\. So there's definitely something
there needing more work\.

\-\- 
You never really learn to swear until you learn to drive\.

**Comment by p5pRT** [comment]
### From zefram@fysh.org

Dave Mitchell wrote&#8203;:
>1\) the test script \(and thus presumably Regexp&#8203;::Grammars\) only worked
>because it was inadvertently exploiting a hole in the "use re eval"
>mechanism\, and that perl 5\.17\.x is correctly refusing to execute the code;

The qr constant overloading behaves in a rather strange way\, compared to
operator overloading\.  After the R&#8203;:G&#8203;:Precursor object has been built up
by the constant\-overload function\, it gets stringified\, invoking the ""
operator overload on R&#8203;:G&#8203;:Precursor\.  The problem is that the resulting
string gets compiled to a regexp object by the core\, not by the module
or the calling program&#8203;: where the re\-eval setting matters\, there's no
way to set it\.

You'd think you could avoid this problem by making R&#8203;:G return a compiled
regexp somewhere instead of a string&#8203;: it can set re\-eval for its explicit
compilation\.  But the core is apparently quite insistent on reducing
the qr\-constant to a string\.  If the constant\-overload function returns
a compiled regexp\, that gets stringified\.  If the "" operator overload
returns a compiled regexp\, that gets stringified\.  If the "" operator
overload returns a ref to an object with more operator overloading\,
that gets stringified\.  qr operator overloading never gets invoked\.

I reckon the qr constant overload should be allowed to return a compiled
regexp\, to be used as\-is rather than stringified\.

\-zefram

**Comment by p5pRT** [comment]
### From @iabyn

On Wed\, Mar 27\, 2013 at 10&#8203;:58&#8203;:23AM \+0000\, Zefram wrote&#8203;:
> I reckon the qr constant overload should be allowed to return a compiled
> regexp\, to be used as\-is rather than stringified\.

Yes\, that's exactly what I've been working on for the last week\.

\-\- 
No matter how many dust sheets you use\, you will get paint on the carpet\.

**Comment by p5pRT** [comment]
### From @iabyn

On Sun\, Mar 10\, 2013 at 12&#8203;:48&#8203;:16AM \+0000\, Dave Mitchell wrote&#8203;:
> However\, I would expect something like the following \*should\* work and be
> safe\, but it doesn't&#8203;:
> 
>     use re 'eval'; return qr/\(?\{ $RGP&#8203;::Success = 1 \}\)$$re/;
> 
> And as you pointed out\,  a 'use re eval' even in the scope of the main
> qr/foo/ doesn't turn off the fatal error\. So there's definitely something
> there needing more work\.

The branch smoke\-me/davem/re\_overload\, currently being smoked and shortly
to be merged into blead\, should fix this\.

If the following change is made to your test script&#8203;:

<details><summary>Inline Patch</summary>

```diff
--- /tmp/rgtest.pl.orig	2013-04-11 10:28:53.879421869 +0100
+++ /tmp/rgtest.pl	2013-04-11 10:31:57.510778815 +0100
@@ -18,7 +18,7 @@
   package RGP;
   use overload q{""} => sub {
     my $re = shift;
-    return '(?{ $RGP::Success = 1 })' . $$re;
+    return qr/(?{ $RGP::Success = 1 })$$re/;
   }, fallback => 1;
 }
 

```
</details>
then it passes both tests when using my new branch. To make it still work
on 5\.16\.x and earlier\, you also need a use re eval&#8203;:

\-    return '\(?\{ $RGP&#8203;::Success = 1 \}\)' \. $$re;
\+    use re 'eval';
\+    return qr/\(?\{ $RGP&#8203;::Success = 1 \}\)$$re/;

which is harmless\, but unnecessary after my fix to blead\.

I haven't tried fixing up Regex&#8203;::Grammars to see if it works\, since it's
too complex for a cursory glance and fix\-up\. I see lots of places in the
code where it returns strings containing '\(?\{'\, but I don't know how
practical it would be to convert all those to return qr//'s instead\.

\-\- 
Gravity is just a theory; teach Intelligent Falling in our schools\!
&nbsp;    http&#8203;://www\.theonion\.com/content/node/39512

**Comment by p5pRT** [comment]
### From zefram@fysh.org

Dave Mitchell wrote&#8203;:
>I haven't tried fixing up Regex&#8203;::Grammars to see if it works\, since it's
>too complex for a cursory glance and fix\-up\. I see lots of places in the
>code where it returns strings containing '\(?\{'\, but I don't know how
>practical it would be to convert all those to return qr//'s instead\.

It only needs a qr// once\, at the top level\.

\-zefram

**Comment by p5pRT** [comment]
### From @iabyn

On Thu\, Apr 11\, 2013 at 10&#8203;:39&#8203;:26AM \+0100\, Dave Mitchell wrote&#8203;:
> On Sun\, Mar 10\, 2013 at 12&#8203;:48&#8203;:16AM \+0000\, Dave Mitchell wrote&#8203;:
> > However\, I would expect something like the following \*should\* work and be
> > safe\, but it doesn't&#8203;:
> > 
> >     use re 'eval'; return qr/\(?\{ $RGP&#8203;::Success = 1 \}\)$$re/;
> > 
> > And as you pointed out\,  a 'use re eval' even in the scope of the main
> > qr/foo/ doesn't turn off the fatal error\. So there's definitely something
> > there needing more work\.
> 
> The branch smoke\-me/davem/re\_overload\, currently being smoked and shortly
> to be merged into blead\, should fix this\.

And now in blead with commit

&nbsp;    e501306eca0fea1cc9fc53e2eb024ad37e85ce59

\-\- 
My get\-up\-and\-go just got up and went\.

**Comment by p5pRT** [comment]
@iabyn - Status changed from 'open' to 'resolved'

**Comment by p5pRT** [comment]
### From @rjbs

\* Dave Mitchell \<davem@&#8203;iabyn\.com> \[2013\-04\-12T10&#8203;:57&#8203;:15\]
> And now in blead with commit
> 
>     e501306eca0fea1cc9fc53e2eb024ad37e85ce59

I'm delighted\, thanks Dave\!

\-\- 
rjbs

**Comment by p5pRT** [comment]
### From @hroncok

Why is this marked as solved? I cannot run Regexp&#8203;::Grammars in 5\.18\. 
Even if I backport all mentioned commits&#8203;:

commit c3923c33af542d8764d5a1e4eb5d7b311f443b89
Author&#8203;:     David Mitchell \<davem@&#8203;iabyn\.com>
AuthorDate&#8203;: Tue Aug 6 16&#8203;:34&#8203;:50 2013 \+0100
Commit&#8203;:     David Mitchell \<davem@&#8203;iabyn\.com>
CommitDate&#8203;: Tue Aug 6 16&#8203;:44&#8203;:12 2013 \+0100

&nbsp;    reparse compile\-time /\(?\{\}\)/ in right scope

\<http&#8203;://www\.nntp\.perl\.org/group/perl\.perl5\.porters/2013/07/msg205393\.htm
l>&#8203;:

commit c30fc27b4df65a43710b25dd1d2a57d78ee2fe33
Author&#8203;: David Mitchell \<davem@&#8203;iabyn\.com>
Date&#8203;:   Wed Jul 31 22&#8203;:41&#8203;:17 2013 \+0100

&nbsp;    Handle /\[\#\]/ and /\[\(?\#\]/ with code blocks

Is there a fix somewhere present and I just don't see it? Thanks

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

Why is this marked as solved? I cannot run Regexp&#8203;::Grammars in 5\.18\. 
Even if I backport all mentioned commits&#8203;:

commit c3923c33af542d8764d5a1e4eb5d7b311f443b89
Author&#8203;:     David Mitchell \<davem@&#8203;iabyn\.com>
AuthorDate&#8203;: Tue Aug 6 16&#8203;:34&#8203;:50 2013 \+0100
Commit&#8203;:     David Mitchell \<davem@&#8203;iabyn\.com>
CommitDate&#8203;: Tue Aug 6 16&#8203;:44&#8203;:12 2013 \+0100

&nbsp;    reparse compile\-time /\(?\{\}\)/ in right scope

\<http&#8203;://www\.nntp\.perl\.org/group/perl\.perl5\.porters/2013/07/msg205393\.htm
l>&#8203;:

commit c30fc27b4df65a43710b25dd1d2a57d78ee2fe33
Author&#8203;: David Mitchell \<davem@&#8203;iabyn\.com>
Date&#8203;:   Wed Jul 31 22&#8203;:41&#8203;:17 2013 \+0100

&nbsp;    Handle /\[\#\]/ and /\[\(?\#\]/ with code blocks

Is there a fix somewhere present and I just don't see it? Thanks

**Comment by p5pRT** [comment]
### From @Leont

On Wed\, Oct 16\, 2013 at 4&#8203;:37 PM\, Miro HronÄok via RT \<
perlbug\-comment@&#8203;perl\.org> wrote&#8203;:

> Why is this marked as solved? I cannot run Regexp&#8203;::Grammars in 5\.18\.
> Even if I backport all mentioned commits&#8203;:
>
> commit c3923c33af542d8764d5a1e4eb5d7b311f443b89
> Author&#8203;:     David Mitchell \<davem@&#8203;iabyn\.com>
> AuthorDate&#8203;: Tue Aug 6 16&#8203;:34&#8203;:50 2013 \+0100
> Commit&#8203;:     David Mitchell \<davem@&#8203;iabyn\.com>
> CommitDate&#8203;: Tue Aug 6 16&#8203;:44&#8203;:12 2013 \+0100
>
>     reparse compile\-time /\(?\{\}\)/ in right scope
>
> \<http&#8203;://www\.nntp\.perl\.org/group/perl\.perl5\.porters/2013/07/msg205393\.htm
> l>&#8203;:
>
> commit c30fc27b4df65a43710b25dd1d2a57d78ee2fe33
> Author&#8203;: David Mitchell \<davem@&#8203;iabyn\.com>
> Date&#8203;:   Wed Jul 31 22&#8203;:41&#8203;:17 2013 \+0100
>
>     Handle /\[\#\]/ and /\[\(?\#\]/ with code blocks
>
> Is there a fix somewhere present and I just don't see it? Thanks
>

CPAN Testers is all green on 5\.18\.1\, as well as 5\.19\.2\+\. Are you sure
you're not accidentally trying to run this on 5\.18\.0?

Leon
